### PR TITLE
ASDPLNG-136: Create Profile for Kerberos Server

### PR DESCRIPTION
**Pull request by billglick**
This is being tested on:
- `wood`

See: https://jira.ncsa.illinois.edu/browse/ASDPLNG-136

This is a port of an existing Puppet module that was used on `its-foreman`. We are skipping any general 'improvements' or refactoring as much as possible. This review is more to look for issues.

The review should be of the whole module, not just the changes in this branch.

### CODE REVIEW

File: manifests/firewall.pp [hunk 1]
```diff
@@ -1,40 +1,35 @@
...
+#
 # @example
 #   include profile_kerberos_server::firewall
-class profile_kerberos_server::firewall {
+class profile_kerberos_server::firewall (
+  String $primary_server,
+) {
 
-#-A INPUT -p tcp -s brick.ncsa.uiuc.edu --dports krb5_prop -j ACCEPT
   firewall { '1000 Kerberos krb5_prop via TCP':
```
Comment (andylytical) [inline manifests/firewall.pp]: Rule number should be less than 900 so it doesn't come after `999 DROP ALL` (from the base firewall profile)

Suggest 200-400 range to indicate these are "service" related.

File: manifests/firewall.pp [hunk 2]
```diff
@@ -1,40 +1,35 @@
...
     dport  => 'krb5_prop',
-    source => 'brick.ncsa.uiuc.edu',
+    source => $primary_server,
     action => 'accept',
-#    provider => 'iptables',
   }
 
-# Don't need kpasswd on slaves, only on master
-#-A INPUT -p all -m multiport --dports kerberos,kpasswd -j ACCEPT
   firewall { '1010 Kerberos via TCP':
```
Comment (andylytical) [inline manifests/firewall.pp]: Rule number should be less than 900 so it doesn't come after `999 DROP ALL` (from the base firewall profile)

Suggest 200-400 range to indicate these are "service" related.

File: manifests/firewall.pp [hunk 3]
```diff
@@ -1,40 +1,35 @@
...
-#      'kpasswd',
     ],
-    action   => 'accept',
-    provider => 'iptables',
+    action => 'accept',
   }
 
-# Don't need kpasswd on slaves, only on master
-#-A INPUT -p all -m multiport --dports kerberos,kpasswd -j ACCEPT
   firewall { '1011 Kerberos via UDP':
```
Comment (andylytical) [inline manifests/firewall.pp]: Rule number should be less than 900 so it doesn't come after `999 DROP ALL` (from the base firewall profile)

Suggest 200-400 range to indicate these are "service" related.

File: manifests/services.pp [hunk 4]
```diff
@@ -1,54 +1,40 @@
...
+) {
 
   ensure_packages($packages_present, {'ensure' => 'present'})
   ensure_packages($packages_absent, {'ensure' => 'absent'})
 
 #NOTE:  This class' configuration of hosts.allow and hosts.deny could conflict with other
 #       definitions for them in other classes.
   file { '/etc/hosts.allow':
-    source => 'puppet:///modules/kerberos_server/etc/hosts.allow',
+    source => "puppet:///modules/${module_name}/etc/hosts.allow",
```
Comment (andylytical) [inline manifests/services.pp:22]: tcpwrappers doesn't exist in RHEL8 (and is deprecated in RHEL7)

Suggest these could be removed right now?!

File: manifests/services.pp [hunk 5]
```diff
@@ -1,54 +1,40 @@
...
     mode   => '0644',
   }
 
   file { '/etc/hosts.deny':
-    source => 'puppet:///modules/kerberos_server/etc/hosts.deny',
+    source => "puppet:///modules/${module_name}/etc/hosts.deny",
```
Comment (andylytical) [inline manifests/services.pp:27]: tcpwrappers doesn't exist in RHEL8 (and is deprecated in RHEL7)

Suggest these could be removed right now?!

File: manifests/services.pp [hunk 6]
```diff
@@ -1,54 +1,40 @@
...
 
   file { '/etc/xinetd.d/krb5-prop':
-    source => 'puppet:///modules/kerberos_server/etc/xinetd.d/krb5-prop',
+    source => "puppet:///modules/${module_name}/etc/xinetd.d/krb5-prop",
     mode   => '0644',
   }
 
   file { '/etc/rsyslog.d/rsyslog-kdc.conf':
-    source => 'puppet:///modules/kerberos_server/etc/rsyslog.d/rsyslog-kdc.conf',
+    source => "puppet:///modules/${module_name}/etc/rsyslog.d/rsyslog-kdc.conf",
```
Comment (andylytical) [inline manifests/services.pp:37]: Does the rsyslog service need to be notified about this new config file?

File: manifests/services.pp [hunk 7]
```diff
@@ -1,54 +1,40 @@
...
 
   file { '/etc/hosts.deny':
-    source => 'puppet:///modules/kerberos_server/etc/hosts.deny',
+    source => "puppet:///modules/${module_name}/etc/hosts.deny",
     mode   => '0644',
   }
 
   file { '/etc/xinetd.d/krb5-prop':
-    source => 'puppet:///modules/kerberos_server/etc/xinetd.d/krb5-prop',
+    source => "puppet:///modules/${module_name}/etc/xinetd.d/krb5-prop",
```
Comment (andylytical) [inline manifests/services.pp:32]: Does xinetd need to be notified?

**Review by andylytical — CHANGES_REQUESTED** [review]
* Firewall rule numbers should be less than 900
  * The default `DROP ALL` rule is 999, there's a good chance higher numbers go after and thus are never reached.
  * Even if puppet sorts these lexigraphically, it's still confusing to maintain.

File: manifests/services.pp (hunk 4 above)
Comment (billglick) [reply manifests/services.pp:22]: Yes. But I'd like to delay functional changes to a later tag release.

File: manifests/services.pp (hunk 5 above)
Comment (billglick) [reply manifests/services.pp:27]: Yes. But I'd like to delay functional changes to a later tag release.

File: manifests/services.pp (hunk 6 above)
Comment (billglick) [reply manifests/services.pp:37]: Yes. But I'd like to delay functional changes to a later tag release.

File: manifests/services.pp (hunk 7 above)
Comment (billglick) [reply manifests/services.pp:32]: Yes. But I'd like to delay functional changes to a later tag release.

File: manifests/services.pp (hunk 6 above)
Comment (andylytical) [reply manifests/services.pp:37]: Should probably be an enhancement request.

File: manifests/firewall.pp (hunk 3 above)
Comment (andylytical) [reply manifests/firewall.pp]: fixed.

File: manifests/services.pp (hunk 7 above)
Comment (billglick) [reply manifests/services.pp:32]: I will create github issues for enhancement requests & create ASDPLNG tickets for each of those.

**Review by andylytical — APPROVED** [review]
Looks good. Thanks for the fixes.

File: manifests/services.pp (hunk 6 above)
Comment (billglick) [reply manifests/services.pp:37]: Will address in https://github.com/ncsa/puppet-profile_kerberos_server/issues/3

File: manifests/services.pp (hunk 7 above)
Comment (billglick) [reply manifests/services.pp:32]: Will address in https://github.com/ncsa/puppet-profile_kerberos_server/issues/3

File: manifests/services.pp (hunk 4 above)
Comment (billglick) [reply manifests/services.pp:22]: Will address in https://github.com/ncsa/puppet-profile_kerberos_server/issues/2

File: manifests/services.pp (hunk 5 above)
Comment (billglick) [reply manifests/services.pp:27]: Will address in https://github.com/ncsa/puppet-profile_kerberos_server/issues/2

**Pull request merged**